service/src/main/kotlin/fi/espoo/evaka/placement/PlacementTermination.kt

```diff
@@ -218,21 +218,34 @@ fun terminateBilledDaycare(
             }
         } else {
             tx.updatePlacementEndDate(placement.id, terminationDate)
+            tx.clearGroupPlacementsAfter(placement.id, terminationDate)
```

Comment:
This looks correct, but I think not enough. For example, below there's `deleteServiceNeedsFromPlacementAfter` which only deletes service needs that are fully after the terminationDate, but does not cut ones that overlap. 

Maybe `tx.updatePlacementEndDate` should be replaced with `updatePlacementEndDate(tx, ...)` which would not only do the simple update but also update/delete all hierarchical descendants (group placements, service needs, maybe something else too). The original update query could even be made private (if it isn't needed e.g. in tests) to make sure it's not used elsewhere, because using it alone breaks invariants. Same would need to be also done for all other functions where placement start and/or end date changes.